### PR TITLE
Don't generate setter for output properties

### PR DESCRIPTION
**Pull request by ogoffart**
Also make it an error to change the value from the interpreter

### CODE REVIEW

File: internal/interpreter/api.rs [hunk 1]
```diff
@@ -1159,13 +1164,33 @@ fn component_definition_properties2() {
...
     assert_eq!(callbacks.len(), 1);
     assert_eq!(callbacks[0], "hello");
+
+    let instance = comp_def.create();
+    assert_eq!(
+        instance.set_property("xreadonly", SharedString::from("XXX").into()),
+        Err(SetPropertyError::AccessDenied)
+    );
+    assert_eq!(instance.get_property("xreadonly"), Ok(Value::String("the value".into())));
+    assert_eq!(instance.get_property("xreadonly"), Ok(Value::String("the value".into())));
```
Comment (tronical) [inline internal/interpreter/api.rs:1182]: Just curious, why test the getter twice?

File: internal/interpreter/api.rs [hunk 2]
```diff
@@ -1035,6 +1035,9 @@ pub enum SetPropertyError {
     /// The property exist but does not have a type matching the dynamic value
     #[error("wrong type")]
     WrongType,
+    /// Attempt to set an output property
+    #[error("access denied")]
+    AccessDenied,
```
Comment (tronical) [inline internal/interpreter/api.rs:1040]: I wonder if there should be different errors, one for attempted access to a private property and one for trying to write to an output property? The former could also produces “NoSuchProperty” instead.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ All notable changes to this project are documented in this file.
    There was a warning about this change in previous versions (#1394)
  - `Window`'s `default-font-size` property is now always set to a non-zero value, provided by 
    either the style or the backend.
+ - In the interpreter, calling set_property or get_property for property of the base no longer work
```
Comment (tronical) [inline CHANGELOG.md:12]:
```suggestion
 - In the interpreter, calling set_property or get_property on properties of the base no longer works.
```

**Review by tronical — APPROVED** [review]
Looks good to me in general. Not entirely sure about the error api.

File: internal/interpreter/api.rs (hunk 2 above)
Comment (ogoffart) [reply internal/interpreter/api.rs:1040]: this is the case already.  Private == NoSuchProperty

File: internal/interpreter/api.rs (hunk 1 above)
Comment (ogoffart) [reply internal/interpreter/api.rs:1182]: bad copy paste

**Comment by ogoffart** [comment]
In order to make the test pass, i decide to change the check from the dymanic_component to the api.rs module (in the interpreter).
Because i realized that there are case where we call these function with internal properties.

File: internal/interpreter/api.rs (hunk 2 above)
Comment (tronical) [reply internal/interpreter/api.rs:1068]: Ahh great

**Review by tronical — APPROVED** [review]

**Pull request merged**